packages/app/src/specs/banners/TrackedWarning.cy.tsx

```diff
@@ -0,0 +1,40 @@
+import faker from 'faker'
+import TrackedWarning from './TrackedWarning.vue'
+import { ref } from 'vue'
+import { defaultMessages } from '@cy/i18n'
+
+describe('<TrackedWarning />', () => {
+  it('calls dismiss when X is clicked', () => {
+    const title = faker.hacker.noun()
+    const message = `
+  # Hello!
+  > This is a **markdown formatted** message!
+  
+  We're going to print out some \`console.log('cool code')\` and see how well it formats inside of our warning.
+  `
+    const bannerId = 'test123'
+    const show = ref(true)
+    const onUpdate = cy.spy()
+    const methods = {
+      'onUpdate:modelValue': (value) => {
+        show.value = value
+        onUpdate()
+      },
+    }
+
+    cy.mount(() => (<div class="p-4"><TrackedWarning
+      data-testid="warning"
+      title={title}
+      message={message}
+      bannerId={bannerId}
+      {...methods}
+    /></div>))
+
+    cy.get('[data-testid=warning]').as('warning')
+
+    cy.get('@warning').should('be.visible')
+    cy.get(`[aria-label=${defaultMessages.components.alert.dismissAriaLabel}`).first().click()
+    cy.wrap(onUpdate).should('be.called')
```

Comment:
This assertion just validates that the event is propagated out of the `TrackedWarning`.   What should be tested is that the settings are saved to the server when the warning is dismissed. See an example for the `TrackedBanner` here: https://github.com/cypress-io/cypress/blob/develop/packages/app/src/specs/banners/TrackedBanner.cy.tsx#L38-L63. 

The `cy.stubMutationResolver` can track when the GraphQL mutation is called.